### PR TITLE
Application by applicant

### PR DESCRIPTION
**Pull request by magnearun**
# \<Description\>

## What

Create endpoint that return applications for a applicant using nationalRegistrationId.
Add name and progress to the response dto for applications

## Why

To be able to list "my" applications in the service portal

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against master before asking for a review

### CODE REVIEW

**Review by sindrig — APPROVED** [review]
LGTM

File: libs/application/templates/driving-lessons/src/lib/DrivingLessonsTemplate.ts [hunk 1]
```diff
@@ -41,6 +41,7 @@ const DrivingLessonsTemplate: ApplicationTemplate<
   Events
 > = {
   type: ApplicationTypes.DRIVING_LESSONS,
+  name: 'Umsókn um ökunám!',
```
Comment (atligudl) [inline libs/application/templates/driving-lessons/src/lib/DrivingLessonsTemplate.ts:44]: !!!

File: libs/service-portal/applications/src/screens/ApplicationList/ApplicationList.tsx [hunk 2]
```diff
@@ -20,15 +26,37 @@ const mockApplications: MockApplication[] = [
...
   return (
     <>
       <Box marginBottom={5}>
         <Typography variant="h1">Umsóknir</Typography>
       </Box>
+      {loading && <ActionCardLoader repeat={3} />}
       <Stack space={2}>
-        {mockApplications.map((application, index) => (
-          <ApplicationCard application={application} key={index} />
+        {mockApplications.map(({ name, date, status, url }, index) => (
```
Comment (atligudl) [inline libs/service-portal/applications/src/screens/ApplicationList/ApplicationList.tsx:41]: Do we still wanna keep these in?

**Review by atligudl — APPROVED** [review]
LGTM

File: libs/service-portal/applications/src/screens/ApplicationList/ApplicationList.tsx (hunk 2 above)
Comment (magnearun) [reply libs/service-portal/applications/src/screens/ApplicationList/ApplicationList.tsx:41]: I will remove them

File: apps/application-system/api/src/app/modules/application/application.service.ts [hunk 3]
```diff
@@ -1,11 +1,11 @@
 import { Inject, Injectable } from '@nestjs/common'
 import { InjectModel } from '@nestjs/sequelize'
+import { FindOptions } from 'sequelize/types'
 import { Logger, LOGGER_PROVIDER } from '@island.is/logging'
-import { ExternalData } from '@island.is/application/core'
 import { Application } from './application.model'
 import { CreateApplicationDto } from './dto/createApplication.dto'
 import { UpdateApplicationDto } from './dto/updateApplication.dto'
-import { ApplicationTypes } from '@island.is/application/core'
+import { ApplicationTypes, ExternalData } from '@island.is/application/core'
```
Comment (arnifreyrs) [inline apps/application-system/api/src/app/modules/application/application.service.ts:8]: is this used?

File: apps/application-system/api/src/app/modules/application/tools/application.serializer.ts [hunk 4]
```diff
@@ -3,42 +3,58 @@ import {
...
+  async serialize(model: Application) {
+    const application = model.toJSON() as BaseApplication
+    const template = await getApplicationTemplateByTypeId(
+      application.typeId as ApplicationTypes,
+    )
+    const helper = new ApplicationTemplateHelper(application, template)
+
+    const dto = plainToClass(ApplicationResponseDto, {
+      ...application,
+      ...helper.getPermittedAnswersAndExternalData(role),
```
Comment (arnifreyrs) [inline apps/application-system/api/src/app/modules/application/tools/application.serializer.ts:53]: you can do `...helper.getPermittedAnswersAndExternalData(template.mapUserToRole([nationalRegistryId])`

but I don't know how to get the `nationalRegistryId` at this specific place

File: apps/application-system/api/src/app/modules/application/application.controller.ts [hunk 5]
```diff
@@ -69,24 +72,78 @@ export class ApplicationController {
...
     if (typeId) {
       return this.applicationService.findAllByType(typeId as ApplicationTypes)
     } else {
       return this.applicationService.findAll()
     }
   }
 
-  @Post()
-  @ApiCreatedResponse({ type: Application })
+  @Get('applicants/:nationalRegistryId/applications')
```
Comment (arnifreyrs) [inline apps/application-system/api/src/app/modules/application/application.controller.ts:90]: you should add tests for these new endpoints in the e2e spec

File: libs/application/core/src/lib/ApplicationTemplateHelper.ts [hunk 6]
```diff
@@ -44,6 +44,12 @@ export class ApplicationTemplateHelper<
     )
   }
 
+  getApplicationProgress(stateKey: string = this.application.state): number {
```
Comment (arnifreyrs) [inline libs/application/core/src/lib/ApplicationTemplateHelper.ts:47]: can you add a test for this as well?

File: libs/application/core/src/types/Application.ts [hunk 7]
```diff
@@ -22,4 +22,6 @@ export interface Application {
   attachments: object
   answers: FormValue
   externalData: ExternalData
+  name?: string
```
Comment (arnifreyrs) [inline libs/application/core/src/types/Application.ts:25]: so this won't be `MessageDescriptor | string` ?

File: libs/service-portal/applications/src/screens/ApplicationList/ApplicationList.tsx [hunk 8]
```diff
@@ -1,34 +1,42 @@
...
+      )}
       <Stack space={2}>
-        {mockApplications.map((application, index) => (
-          <ApplicationCard application={application} key={index} />
+        {applications?.map((application: Application) => (
+          <ApplicationCard
+            key={application.id}
+            name={application.name || application.typeId}
+            date={format(new Date(application.modified), 'MMMM')}
+            status={application.state === 'approved'}
```
Comment (arnifreyrs) [inline libs/service-portal/applications/src/screens/ApplicationList/ApplicationList.tsx:36]: we will have to improve this later, I think that the `status` prop should instead be called `isCompleted` and be:

```
isCompleted={application.progress === 1}
```

**Review by arnifreyrs — CHANGES_REQUESTED** [review]
Some minor changes, otherwise looks great!

File: libs/service-portal/applications/src/screens/ApplicationList/ApplicationList.tsx (hunk 8 above)
Comment (magnearun) [reply libs/service-portal/applications/src/screens/ApplicationList/ApplicationList.tsx:36]: agree

File: libs/application/core/src/types/Application.ts (hunk 7 above)
Comment (magnearun) [reply libs/application/core/src/types/Application.ts:25]: No, this is the application after it is returned from the backend, right ?

File: libs/application/core/src/lib/ApplicationTemplateHelper.ts (hunk 6 above)
Comment (magnearun) [reply libs/application/core/src/lib/ApplicationTemplateHelper.ts:47]: will do

File: apps/application-system/api/src/app/modules/application/application.controller.ts (hunk 5 above)
Comment (magnearun) [reply apps/application-system/api/src/app/modules/application/application.controller.ts:90]: will do

File: libs/application/core/src/types/Application.ts (hunk 7 above)
Comment (arnifreyrs) [reply libs/application/core/src/types/Application.ts:25]: yes, I was just a bit lost about where the actual message descriptor resides. but this is fine

**Review by arnifreyrs — DISMISSED** [review]
great stuff

**Review by arnifreyrs — APPROVED** [review]

**Review by atligudl — DISMISSED** [review]

**Review by eirikurn — APPROVED** [review]

**Review by atligudl — APPROVED** [review]

**Pull request merged**